backend/oasst_backend/api/v1/tasks.py

```diff
@@ -25,22 +50,17 @@
 )  # work with Union once more types are added
 def request_task(
     *,
-    db: Session = Depends(deps.get_db),
-    api_key: APIKey = Depends(deps.get_api_key),
+    task_context_factory: TaskContextFactory = Depends(get_task_context_factory),
     request: protocol_schema.TaskRequest,
 ) -> Any:
     """
     Create new task.
     """
-    api_client = deps.api_auth(api_key, db)
-
     try:
-        pr = PromptRepository(db, api_client, client_user=request.user)
-        pr.ensure_user_is_enabled()
-
-        tm = TreeManager(db, pr)
-        task, message_tree_id, parent_message_id = tm.next_task(desired_task_type=request.type, lang=request.lang)
-        pr.task_repository.store_task(task, message_tree_id, parent_message_id, request.collective)
+        ctx: TaskContext = task_context_factory(request.user)
```

Comment:
It's a shame `.user` is not common between all requests. This `task_context_factory` is workable but a little more difficult than if it were to just be injected automatically and be user "aware". Just musing

Reply:
Yeah, I also wanted to inject the object but had to go with this indirect factory because of the user.